devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/QuarkusDev.java

```diff
@@ -262,26 +275,49 @@ private QuarkusDevModeLauncher newLauncher() throws Exception {
 
         final ApplicationModel appModel = extension().getApplicationModel(LaunchMode.DEVELOPMENT);
         final Set<GACT> projectDependencies = new HashSet<>();
+
         addSelfWithLocalDeps(project, builder, new HashSet<>(), projectDependencies, true);
 
+        //look for an application.properties
+        Set<Path> resourceDirs = new HashSet<>();
+        for (SourceDir resourceDir : appModel.getApplicationModule().getMainSources().getResourceDirs()) {
+            resourceDirs.add(resourceDir.getOutputDir());
+        }
+
+        Set<ArtifactKey> configuredParentFirst = QuarkusBootstrap.createClassLoadingConfig(PathsCollection.from(resourceDirs),
+                QuarkusBootstrap.Mode.DEV, Collections.emptyList()).parentFirstArtifacts;
+
+        Set<ArtifactKey> parentFirstArtifactKeys = new HashSet<>(configuredParentFirst);
+        for (io.quarkus.maven.dependency.ResolvedDependency resolvedDependency : appModel.getDependencies()) {
+            for (Path path : resolvedDependency.getResolvedPaths()) {
+                if (Files.exists(path) && path.getFileName().endsWith(ArtifactCoords.TYPE_JAR)) {
+                    extractParentFirstArtifactKey(parentFirstArtifactKeys, path.toFile());
+                }
+            }
+        }
+
         for (io.quarkus.maven.dependency.ResolvedDependency artifact : appModel.getDependencies()) {
             //we only use the launcher for launching from the IDE, we need to exclude it
             if (artifact.getGroupId().equals("io.quarkus")
                     && artifact.getArtifactId().equals("quarkus-ide-launcher")) {
                 continue;
             }
+            GACT key = new GACT(artifact.getGroupId(), artifact.getArtifactId());
             if (!projectDependencies
-                    .contains(new GACT(artifact.getGroupId(), artifact.getArtifactId()))) {
+                    .contains(key)) {
                 artifact.getResolvedPaths().forEach(p -> {
-                    if (Files.exists(p)) {
+                    File file = p.toFile();
+                    if (file.exists()) {
                         if (artifact.getGroupId().equals("io.quarkus")
                                 && artifact.getArtifactId().equals("quarkus-class-change-agent")) {
-                            builder.jvmArgs("-javaagent:" + p.toFile().getAbsolutePath());
+                            builder.jvmArgs("-javaagent:" + file.getAbsolutePath());
                         } else {
-                            addToClassPaths(builder, p.toFile());
+                            if (parentFirstArtifactKeys.contains(key) && filesIncludedInClasspath.add(file)) {
```

Comment:
```suggestion
                            if (appModel.getParentFirst().contains(key) && filesIncludedInClasspath.add(file)) {
```
and remove the rest of the changes?

Reply:
We just need the fix Stuart introduced which looks for parent first artifact in ˋapplication.properties` file. But most of the rest can be simplified